spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -2529,4 +2529,21 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
         spark.sql("SELECT array_contains((CASE WHEN _2 =_3 THEN array(_4) END), _4) FROM t1"));
     }
   }
+  test("array_remove") {
+    Seq(true, false).foreach { dictionaryEnabled =>
+      withTempDir { dir =>
+        val path = new Path(dir.toURI.toString, "test.parquet")
+        makeParquetFileAllTypes(path, dictionaryEnabled, 10000)
+        spark.read.parquet(path.toString).createOrReplaceTempView("t1");
+        checkSparkAnswerAndOperator(
+          sql("SELECT array_remove(array(_2, _3,_4), _2) from t1 where _2 is null"))
+        checkSparkAnswerAndOperator(
+          sql("SELECT array_remove(array(_2, _3,_4), _3) from t1 where _3 is not null"))
+        checkSparkAnswerAndOperator(sql(
+          "SELECT array_remove(case when _2 = _3 THEN array(_2, _3,_4) ELSE null END, _3) from t1"))
+        checkSparkAnswerAndOperator(sql(
+          "SELECT array_remove(case when _2 = _3 THEN array(_2, _3,_4) ELSE array() END, _3) FROM t1"))
```

Comment:
Getting error for empty array()

```
- array_remove *** FAILED *** (5 seconds, 932 milliseconds)
  Expected only Comet native operators, but found Project.
  plan: *(1) Project [array_remove(CASE WHEN (cast(_2#1 as smallint) = _3#2) THEN array(cast(_2#1 as int), cast(_3#2 as int), _4#3) ELSE [] END, cast(_3#2 as int)) AS array_remove(CASE WHEN (_2 = _3) THEN array(_2, _3, _4) ELSE array() END, _3)#85]
  +- *(1) CometColumnarToRow
     +- CometScan parquet [_2#1,_3#2,_4#3] Batched: true, DataFilters: [], Format: CometParquet, Location: InMemoryFileIndex(1 paths)[file:/Users/jatin/Documents/open-source/datafusion-comet/spark/target/..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<_2:tinyint,_3:smallint,_4:int> (CometTestBase.scala:184)
 ```
 
 @andygrove

Reply:
I looked into this and the issue is that Comet does not support literal arrays, so does not support `array()`. I think we will need to skip the empty array test.

@parthchandra Do you have any other ideas for testing this?

Reply:
I have an idea for how to test this and how to get more coverage on all array expression tests. I filed https://github.com/apache/datafusion-comet/issues/1269 to track

Reply:
See https://github.com/apache/datafusion-comet/pull/1270